CorsixTH/Src/th_gfx_sdl.cpp

```diff
@@ -1090,10 +1099,32 @@ void sprite_sheet::draw_sprite(render_target* pCanvas, size_t iSprite, int iX,
     }
   }
 
+  if (effect == animation_effect::glowing) {
+    // Let's convert SDL Ticks to Game Ticks
+    uint32_t tick = SDL_GetTicks() / usertick_period_ms;
```

Comment:
I expect (but haven't confirmed) that the desired behaviour would have the speed of the affect influenced by the game speed, that is according to the rate that animation.tick is called, rather than SDL_Ticks.

@mugmuggy do you happen to know off hand?

Reply:
I've updated the code to track world ticks on animation_manager. I'm not entirely sure if I've done it correctly. It seems the lua AnimationManager is not actually hooked up to the C++ one, but instead the lua anims interface seems to be the place where we call through to the c++ animation_amanger.

In any event, the effect now seems to pause with the rest of the game.